crates/wasm-mutate-stats/src/bin/wasm-mutate-stats.rs

```diff
@@ -0,0 +1,327 @@
+                        wasm,
+                        rotations: 0,
+                        seed: 0,
+                        hashes: HashMap::new(),
+                        configs: self
+                            .compiler_configs
+                            .iter()
+                            .map(|c| {
+                                (
+                                    c.join(""),
+                                    ConfigReport {
+                                        hashes: HashMap::new(),
+                                    },
+                                )
+                            })
+                            .collect(),
+                        errors: Vec::new(),
+                    };
+                    let serialized = serde_json::to_string(&newreport).unwrap();
+                    f.write(serialized.as_bytes())?;
+                    Ok(newreport)
+                })
+            })
+    }
+
+    fn save_report(&self, report: &Report, name: &String) {
+        let created = File::create(format!("{}.report.json", name));
+        created
+            .and_then(|mut f| {
+                let serialized = serde_json::to_string(report).unwrap();
+                f.write(serialized.as_bytes())
+            })
+            .expect("Report could not be written");
+    }
+
+    fn run_worker(&self, start: usize, end: usize) {
+        let mut wasmmutate = WasmMutate::default();
+        let mut corpus = self.corpus[start..end].to_vec();
+
+        loop {
+            match corpus.pop() {
+                Some((name, data)) => {
+                    println!("Wasm input {}", name);
+                    let mut wasm = data;
+
+                    let mut report = self.load_report(&name, wasm.clone()).unwrap();
+
+                    wasmmutate.seed(report.seed);
+                    let mut rng = SmallRng::seed_from_u64(report.seed);
+                    // mutate the initial wasm as many rotations are in the loaded report
+                    print!("Reaching previous state...");
+                    for _ in 0..report.rotations {
+                        match wasmmutate.run_with_rng(&wasm, &mut rng) {
+                            Ok(mutated) => {
+                                wasm = mutated;
+                            }
+                            Err(_) => { /* bypass */ }
+                        }
+                    }
+                    println!("Reached");
+                    while report.rotations <= ROTATIONS {
+                        report.rotations += 1;
+                        match wasmmutate.run_with_rng(&wasm, &mut rng) {
+                            Ok(mutated) => {
+                                let mut validator = wasmparser::Validator::new();
+                                match validator.validate_all(&mutated) {
+                                    Ok(_) => {
+                                        match self.compile(&mutated, &name, report.rotations) {
```

Comment:
Do we want to compile in this loop? That means we are measuring how many Wasm modules we can unique mutate *and compile* in a given amount of time, not just how many unique Wasm modules we can mutate in a given amount of time.

I think we should have two separate phases:

1. The first phase is spent solely inside `wasm-mutate`, creating as many mutations as we can within the time/count limit and saving them to disk. *Not* hashing them or anything like that.
2. After the time/count limit is reached, then we do a second post-processing step that 
   * hashes the Wasm files and determines how many are unique
   * compiles the Wasm files to object files with Wasmtime (with and without optimizations)
   * hashes the object files and determines how many of these are unique